### PR TITLE
Add np.gradient

### PR DESCRIPTION
**Pull request by clemisch**
I added support for `numpy.gradient`. There are a couple of issues with this PR:

1. It only supports the `axis` keyword
2. Need for auxiliary function `_gradient_along_axis`
3. The tests don't succeed

I am interested in your opinion about 1 and 2. 

Concerning 3, I see two problems:
3.1 `onp.gradient` only accepts arrays where the regarding axes are at least of length 2. This is not the case in the existing testing shapes `nonempty_nonscalar_array_shapes`.
3.2 `onp.gradient` returns a list of arrays. Even if I change `nonempty_nonscalar_array_shapes` to be large enough everywhere, this causes `self.assertTrue(hasattr(y, '__array__') or onp.isscalar(y))` to fail. 

This snippet at least confirms the functionality of the code:

```python
import numpy as onp
import jax
import jax.numpy as np

x = onp.random.rand(20, 30, 10, 15).astype(onp.float32)
y = np.array(x)

for axis in [None, 0, 1, 2, -1, -2, (0, 1), (0, -1), (-2, -1), (1, 3), (3, 1)]:
    xg = onp.array(onp.gradient(x, axis=axis))
    yg = np.gradient(y, axis=axis)

    assert onp.allclose(xg, yg)
```

### CODE REVIEW

**Comment by clemisch** [comment]
@googlebot I signed it!
